### PR TITLE
packing script tweak and registration fixes

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: util.py [hunk 1]
```diff
@@ -39,6 +39,24 @@
 import subprocess
 
 
+def is_oslinfo_valid():
+    tool_dir, shader_diretories = get_osl_search_paths()
```
Comment (dictoon) [inline util.py]: Typo: `shader_directories`

File: util.py [hunk 2]
```diff
@@ -39,6 +39,24 @@
...
+    osl_info_path = os.path.join(tool_dir, 'oslinfo.exe')
+    if not os.path.exists(osl_info_path):
+        return False
+
+    return True
+
+
+def is_appleseed_cli_valid():
+    appleseed_exe = os.path.join(get_appleseed_bin_dir(), 'appleseed.cli.exe')
```
Comment (dictoon) [inline util.py]: Won't work on Linux and macOS.

File: __init__.py [hunk 3]
```diff
@@ -72,9 +72,9 @@ def register():
 
 
 def unregister():
-    bpy.utils.unregister_module(__name__)
     ui.unregister()
     export.unregister()
     operators.unregister()
     properties.unregister()
     preferences.unregister()
+    bpy.utils.unregister_module(__name__)
```
Comment (dictoon) [inline __init__.py]: If we register it last, we should unregister it first (unless there are good reasons to do otherwise, in which case they must be documented).

File: docs/panels/material/light_material.rst [hunk 4]
```diff
@@ -1,2 +1,26 @@
...
+- Radiance Multiplier:
+	- This is a multiplier to the Radiance value.  You can use this to increase the intensity of the light
+- Exposure:
+	- This is another way of increasing the intensity of the lamp, only this one works with exposure.  Increasing the number by one doubles the strength of the emission
+- Cast Indirect Light:
+	- Defines whether the light will contribute to indirect lighting in addition to direct light.  Turning this off may increase convergence speed but lower the realism of the lighting.
+- Importance Multiplier:
+	- This raises or lowers the likelihood of this light being chosen for a direct sample from a surface hit point.  Changing this may help when the light is not being sampled often enough, however it is usually best left alone.
+- Light Near Start:
+	- This moves the sample point away form the geometric surface of the light.  This can help avoid fireflies when a bright light is near another surface.
```
Comment (dictoon) [inline docs/panels/material/light_material.rst]: This is not quite correct: The distance is measured on the ray, not along the geometric normal. Not sure how to explain this in layman terms though :)

File: docs/panels/material/light_material.rst [hunk 5]
```diff
@@ -1,2 +1,26 @@
...
+	- This selects one of two different emission profiles:
+		- Diffuse EDF:
+		- Cone EDF:
+			- This profile allows you to change the angle of distribution, allowing you to create a pseudo spotlight light distribtion from a mesh lamp
+- Cone EDF Angle (*only applicable to Cone EDF profile*):
+	- This defines the angle of the light emission for the Cone EDF profile
+- Radiance Multiplier:
+	- This is a multiplier to the Radiance value.  You can use this to increase the intensity of the light
+- Exposure:
+	- This is another way of increasing the intensity of the lamp, only this one works with exposure.  Increasing the number by one doubles the strength of the emission
```
Comment (dictoon) [inline docs/panels/material/light_material.rst]: Some descriptions end with a point, some others not. Uniformize?

File: util.py (hunk 2 above)
Comment (dictoon) [reply util.py]: To be honest, I'm not sure these checks are worth the added complexity. I'd remove them.

File: __init__.py (hunk 3 above)
Comment (jdent02) [reply __init__.py]: I mentioned on discord that unregistering the module first is what was causing all the unregistration errors with the other classes.  Moving it to the end fixed that.

Comment (dictoon) [reply __init__.py]: Ok, thanks. I must have missed that. A short comment explaining the issue would be helpful.

**Comment by jdent02** [comment]
Fixed.  I'm gonna rebase before I leave for work.

**Comment by dictoon** [comment]
> Fixed. I'm gonna rebase before I leave for work.

Fantastic, thanks!

File: util.py [hunk 6]
```diff
@@ -82,11 +82,15 @@ def get_osl_search_paths():
...
 
+    nodes = []
+
     if not get_appleseed_bin_dir():
         print("[appleseed] WARNING: Path to appleseed's binary directory not set: rendering and OSL features will not be available.")
-        return []
+        return nodes
 
-    nodes = []
+    if not is_oslinfo_valid():
```
[the remark this answers is not in the document]
Comment (jdent02) [reply util.py]: Merge conflict whoopsie.  I can fix it this evening or if you want to do it go ahead.
